pkg/operator/render/options/config.go

```diff
@@ -57,29 +66,145 @@ type TemplateData struct {
 	FileConfig
 }
 
-func (c RenderedManifests) ListManifestOfType(gvk schema.GroupVersionKind) []RenderedManifest {
+func (renderedManifests RenderedManifests) ListManifestOfType(gvk schema.GroupVersionKind) []RenderedManifest {
 	ret := []RenderedManifest{}
-	for i := range c {
-		obj, err := c[i].GetDecodedObj()
+	for i := range renderedManifests {
+		obj, err := renderedManifests[i].GetDecodedObj()
 		if err != nil {
-			klog.Warningf("failure to read %q: %v", c[i].OriginalFilename, err)
+			klog.Warningf("failure to read %q: %v", renderedManifests[i].OriginalFilename, err)
 			continue
 		}
 		if obj.GetObjectKind().GroupVersionKind() == gvk {
-			ret = append(ret, c[i])
+			ret = append(ret, renderedManifests[i])
 		}
 	}
 
 	return ret
 }
 
+func (renderedManifests RenderedManifests) GetManifest(gvk schema.GroupVersionKind, namespace, name string) *RenderedManifest {
```

Comment:
Why does this return a pointer when previously we were returning a fresh slice of the objects? Does it need to be a pointer?

Reply:
I think I would go for an error and return the errors we find along the way as well, if an operator chooses to ignore the error, thats on them to decide how to handle it, but if they require the manifest, including as much error detail in the return seems useful IMO